src/lsm/lsm_worker.c

```diff
@@ -93,11 +92,13 @@ __lsm_worker(void *arg)
 
 	cookie = (WT_LSM_WORKER_ARGS *)arg;
 	session = cookie->session;
-	conn = S2C(session);
 
 	entry = NULL;
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
-	    F_ISSET(cookie, WT_LSM_WORKER_RUN)) {
+	for (;;) {
+		WT_BARRIER();			/* Don't cache the run flag. */
```

Comment:
Do we use this scheme anywhere else? I don't understand why this check is unique.